Домашнее задание 23 спринта

DaniilKosmachev opened the pull request.

DaniilKosmachev committed:
ДЗ 23, 1 этап:
Реализована иконка приложения

DaniilKosmachev committed:
ДЗ 23, 2 этап

DaniilKosmachev committed:
ДЗ 23, 3 этап

DaniilKosmachev committed:
ДЗ 23, 4 этап

DaniilKosmachev committed:
ДЗ 23, 5 этап

DaniilKosmachev committed:
ДЗ 23 спринта, 6 этап

DaniilKosmachev committed:
ДЗ 23 спринта, 7 этап

DaniilKosmachev committed:
ДЗ 23 спринта, 8 этап

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/edit_playlist/fragment/EditPlaylistFragment.kt:
```diff
@@ -0,0 +1,136 @@
+
+        binding.createB.setOnClickListener {
+            var name = binding.nameET.text.toString()
+            var description = binding.descriptionET.text.toString()
```
Зачем тут var? можно просто val

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/edit_playlist/fragment/EditPlaylistFragment.kt:
```diff
@@ -0,0 +1,136 @@
+            if (imageUri != null) {
+                saveImageToPrivateStorage(imageUri!!, name)
+            }
+            if (!name.isNullOrEmpty()) {
```
name никогда не будет null, т.к. у нее тип String не нулабельный

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/edit_playlist/fragment/EditPlaylistFragment.kt:
```diff
@@ -0,0 +1,136 @@
+                saveImageToPrivateStorage(imageUri!!, name)
+            }
+            if (!name.isNullOrEmpty()) {
+                viewModel.editNewPlaylist(receivedPlaylist!!.id!!, name, description, imageUri.toString())
```
Опасный вызов receivedPlaylist!!.id!!

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/edit_playlist/fragment/EditPlaylistFragment.kt:
```diff
@@ -0,0 +1,136 @@
+        }
+
+
+        if (!receivedPlaylist!!.uri.equals("null")) binding.playlistImageIV.apply {
```
receivedPlaylist!!.uri.equals("null") опасный вызов.
рекомендую переписать этот код!

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/edit_playlist/view_model/EditPlaylistViewModel.kt:
```diff
@@ -0,0 +1,21 @@
+    private var playlistsInteractor: PlaylistsInteractor
+): CreatePlaylistFragmentViewModel(playlistsInteractor) {
+
+    override var dbJob: Job? = null
```
В этой переменной нет никакого смысла, т.к. она нигде не используется. Ты просто сохраняешь в нее Job из  viewModelScope.launch и дальше с этим ничего не делаешь.

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/playlist/fragment/PlaylistFragment.kt:
```diff
@@ -0,0 +1,228 @@
+
+    private fun sharePlaylist() {
+        if (!tracksInPlaylist.isNullOrEmpty()) {
+            var message =
```
Формирование строк для отображения это задача PlaylistFragmentViewModel а не фрагмента, перенести код в VM

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/playlist/fragment/PlaylistFragment.kt:
```diff
@@ -0,0 +1,228 @@
+
+    private fun showDialogAlet(track: Track) {
+        informationDialog = MaterialAlertDialogBuilder(requireContext()).apply {
+            setTitle("Хотите удалить трек?")
```
Вынести в строковые константы.

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/playlist/view_model/PlaylistFragmentViewModel.kt:
```diff
@@ -0,0 +1,74 @@
+        return _StatusUpdatedPlaylist
+    }
+
+    var dbJob: Job? = null
```
Эту переменную можно удалить, ее наличие никак не скажется на работу приложения.
Рекомендую почитать про корутины и сущность Job более подробно. Возможно есть недопонимание как это работает и для чего нужна Job

DaniilKosmachev commented:
Почитал, понял, по сути корутина возвращает объект Job. С помощью тех строк, которые были у меня написаны можно было управлять работой именно той корутины, с помощью которой проинициализирована та или иная Job. Например, если CoroutineStart.LAZY, то в дальнейшем корутина запуститься только тогда когда будет вызов job.start()

DaniilKosmachev committed:
ДЗ 23 спринта, 9 этап, внесение правок
Добавлена обработка нажатия на кнопку backButton на экране плейлиста и редактирования плейлиста
Исправил баг с темной темой (переписал lateinit на ? = null)
Убрал Job'ы лишние
Поправил создание новых плейлистов (если uri картинки null, то записывается null, а не null.toString())
Исправлено поведение фрагмента при создании и редактировании фрагмента (layout подстраивается под фокус в edittext) - теперь видно что вносит юзверь

DaniilKosmachev committed:
ДЗ 23 спринта, этап 10
Установлена правильная сортировка для треков в плейлисте

DaniilKosmachev committed:
ДЗ 23 спринта, этап 11
Поправил верстки экранов Новый плейлист и Плейлист

DaniilKosmachev committed:
ДЗ 23 спринта, этап 12
Поправил верстки экранов Новый плейлист и Плейлист, версия 2

DaniilKosmachev committed:
ДЗ 23 спринта, этап 12
Поправил верстки экранов Новый плейлист и Плейлист, версия 3

DaniilKosmachev committed:
ДЗ 23 спринта, этап 13
